typescript-loader/loader.js

```diff
@@ -0,0 +1,46 @@
+// 💡 It's probably a good idea to put this loader near the end of the chain.
+
+import path from 'node:path';
+
+import { transform } from 'esbuild';
+
+
+const tsExts = new Set([
+  '.tsx',
+  '.ts',
+  '.mts',
+  '.cts',
+]);
+
+export async function resolve(specifier, context, nextResolve) {
+  const ext = path.extname(specifier);
+
+  const resolution = nextResolve(specifier); // No need to await nextResolve (yet)
+
+  if (!tsExts.has(ext)) { return resolution; } // File is not ts, so step aside
```

Comment:
```suggestion
  if (!tsExts.has(ext)) { return nextResolve(specifier); } // File is not ts, so step aside
```

I gather that your code relies on `nextResolve` returning a promise, and that’s why it’s not throwing for TypeScript files (yet?). Wouldn’t it eventually throw when it resolves?

Perhaps it would be better (and easier to understand) if we simply avoid calling `nextResolve` for a specifier that this loader is going to handle.

Reply:
Right but hanging on to an unsettled promise is a bit advanced for new users to follow, and seems unnecessary in this case. Why not just avoid calling `nextResolve` unless we know we need to?

Reply:
Hmm? We always need to call `nextResolve` in all scenarios here. The conditional bit is whether we need it to settle before the hook returns.

The alternative is something like:

```
if (isNotTypescript) { return nextResolve(…); }

const { url } = await nextResolve(…);

return {
  // …
  url,
};
```

Reply:
Okay. I still think it would be more readable to just `await` `nextResolve` where it’s called, rather than hanging onto a handle and awaiting it later. Something like:

```js
// Resolve this specifier to a URL, whether or not it's TypeScript
const resolution = await nextResolve(...)
```

Reply:
Sure, I can do. Incoming